astropy/table/serialize.py

```diff
@@ -128,7 +128,9 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
             new_name = name + '.' + data_attr
 
         if not has_info_class(data, MixinInfo):
-            new_cols.append(Column(data, name=new_name, **info))
+            # We initialize as MaskedColumn so that we keep a possible mask.
+            # If there is no mask, this will behave as a regular column.
+            new_cols.append(MaskedColumn(data, name=new_name, **info))
```

Comment:
I'm hesitant to make this global change and always go to `MaskedColumn` here. I've just had such a long history of being surprised by MaskedArray vs np.array, so I'd prefer to keep the original behavior for the 99% use-case of not-masked attributes.

What about checking for a `mask` attribute first and then `np.any(data.mask)`:
```
col_cls = MaskedColumn if (hasattr(data, 'mask') and np.any(data.mask)) else Column
new_cols.append(col_cls(data, name=new_name, **info))
```

Reply:
Nominally, this is of course what `MaskedColumnInfo` does but I agree initialization itself could cause problems too, so I've changed it to your suggestion. With that, the comment I inserted could be removed since the code became obvious, so it clearly is the better way!